tests/tensor_search/test_delete_documents.py

```diff
@@ -1,271 +1,271 @@
+        cls.text_index_with_random_model = text_index_with_random_model.name
 
-    def _delete_testing_indices(self):
-        for ix in [self.index_name_1]:
-            try:
-                tensor_search.delete_index(config=self.config, index_name=ix)
-            except IndexNotFoundError as s:
-                pass
+    def setUp(self) -> None:
+        self.clear_indexes(self.indexes)
+
+        # Any tests that call add_documents, search, bulk_search need this env var
+        self.device_patcher = patch.dict(os.environ, {"MARQO_BEST_AVAILABLE_DEVICE": "cpu"})
+        self.device_patcher.start()
+
+    def tearDown(self) -> None:
+        self.device_patcher.stop()
 
     def test_delete_documents(self):
         # first batch:
         add_docs_caller(
-            config=self.config, index_name=self.index_name_1,
+            config=self.config, index_name=self.text_index_with_random_model,
             docs=[
                 {"f1": "cat dog sat mat", "Sydney": "Australia contains Sydney"},
                 {"Lime": "Tree tee", "Magnificent": "Waterfall out yonder"},
-            ], auto_refresh=True)
-        count0_res = requests.post(
-            F"{self.endpoint}/{self.index_name_1}/_count",
-            timeout=self.config.timeout,
-            verify=False
-        ).json()["count"]
+            ], tensor_fields=["f1"])
+
+        count0_res = self.monitoring.get_index_stats_by_name(self.text_index_with_random_model).number_of_documents
+
         add_docs_caller(
-            config=self.config, index_name=self.index_name_1,
+            config=self.config, index_name=self.text_index_with_random_model,
             docs=[
                 {"hooped": "absolutely ridic", "Darling": "A harbour in Sydney", "_id": "455"},
                 {"efg": "hheeehehehhe", "_id": "at-at"}
-            ], auto_refresh=True)
-        count1_res = requests.post(
-            F"{self.endpoint}/{self.index_name_1}/_count",
-            timeout=self.config.timeout,
-            verify=False
-        ).json()["count"]
-        marqo.tensor_search.tensor_search.delete_documents(config=self.config, index_name=self.index_name_1, doc_ids=["455", "at-at"],
-                                                           auto_refresh=True)
-        count_post_delete = requests.post(
-            F"{self.endpoint}/{self.index_name_1}/_count",
-            timeout=self.config.timeout,
-            verify=False
-        ).json()["count"]
+            ], tensor_fields=["f1"])
+
+        count1_res = self.monitoring.get_index_stats_by_name(self.text_index_with_random_model).number_of_documents
+
+        marqo.tensor_search.tensor_search.delete_documents(config=self.config, index_name=self.text_index_with_random_model, doc_ids=["455", "at-at"])
+
+        count_post_delete = self.monitoring.get_index_stats_by_name(self.text_index_with_random_model).number_of_documents
+
         assert count_post_delete == count0_res
+        assert count1_res == count0_res + 2
```

Comment:
No need to change these, but do note that we're replacing these with `self.assert` to get a better output when it fails

Reply:
Got this! Will use `self.assert` in future tests